### ISSUE TITLE
Monkeypox

### ISSUE THREAD

**Issue by fanninpm** [usafsam/mad_river_wf#31]
Most of the current workflow should be compatible with the references for hMPXV. However, pangolin seems to be exclusive to SARS-CoV-2.

The URL for the iVar GFF is https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/014/621/545/GCF_014621545.1_ASM1462154v1/GCF_014621545.1_ASM1462154v1_genomic.gff.gz.

**Comment by friesac** [comment]
Yeah, Nextclade should call the clades for us and pangolin can be removed for hmpxv.

**Comment by fanninpm** [comment]
Code changes to make:

- `modules/local/bbtools.nf`:
  - (in general) possibly tweaking a few parameters
  - `BBMAP_ALIGN` (and `BBMAP_INDEX`): find a way to store the index on-disk instead of in-memory
- `modules/local/ivar.nf`: perhaps tweaking a few parameters related to `ivar` and `samtools`
- `modules/local/misc.nf`:
  - `DATA_PREP`:
    - make `--schema-name` argument customizable
    - in the future, use `edirect` to download iVar GFF (if the GFF straight from NCBI gives better iVar results than the GFF from nextclade_data)
  - `LINEAGE_EXCEL`: make pangolin input optional
  - `SPIKE_GENE_COVERAGE`: see #10 
  - `TRAFFIC_LIGHT_PLOT`: in the future, explain what it represents
- `modules/local/nextclade.nf`: make name and reference (accession) customizable
- `modules/local/vadr.nf`: move `params.vadr_{mdir,options,reference,trim_options}` out to `main.nf` or a subworkflow
- `main.nf`: implement logic for handling SARS-CoV-2 vs. hMPXV

**Comment by friesac** [comment]
> * `modules/local/misc.nf`:
>   
>   * `DATA_PREP`:
>     
>     * make `--schema-name` argument customizable
>     * in the future, use `edirect` to download iVar GFF (if the GFF straight from NCBI gives better iVar results than the GFF from nextclade_data)
>   * `LINEAGE_EXCEL`: make pangolin input optional

Yeah, the `LINEAGE_EXCEL` file would not need to be created, or anything like it, for this initial conversion of taking fastq data and turning it into a consensus fasta. 

>   * `SPIKE_GENE_COVERAGE`: see [Generalize `SPIKE_GENE_COVERAGE` to cover all genes #10](https://github.com/usafsam/mad_river_wf/issues/10)

Can we just ignore edits/improvements to `SPIKE_GENE_COVERAGE` process while trying to get to hmpxv conversion?

**Comment by fanninpm** [comment]
As discussed today, we should name these primers something like "hMPXV_YSPH".
